### ISSUE TITLE
Better Formatting of Large JSON Objects in expected/actual Reports

### ISSUE THREAD

**Issue by inadarei** [substack/tape#419]
Couple Issues with the current formatting:

1. The threshold set implicitly or explicitly may be insufficient, in which case the mis-matched value could be swallowed by an un-expanded portion of `{object}` literal.
2. Even when the threshold is set high-enough, the output is a single-line string that is quite messy and unreadable

Case in point:

```
---
    operator: deepEqual
    expected: |-
      { 'h:link': [ { template: { contentType: 'application/x-www-form-urlencoded', fields: { orderNumber: [Object], productCode: [Object], quantity: [Object] } }, uri: 'http://api.x.io/orders/42/items', label: 'Add Item', action: 'append' } ] }
    actual: |-
      { 'h:link': [ { template: { contentType: 'application/x-www-form-urlencoded', fields: { orderNumber: [Object], productCode: [Object], quantity: [Object] } }, uri: 'http://api.x.io/orders/42/items', label: 'Add Item', action: 'append' } ] }
    at: Test.test (/Users/irakli/Dropbox/Projects/node/kokua/test/unit/siren-reverse.js:77:5)
    stack: |-
      Error: Converted properly
```

can be auto-formatted more legibly as:

```
operator: deepEqual
    expected: |-
      {
  "h:link": [
    {
      "template": {
        "contentType": "application/x-www-form-urlencoded",
        "fields": {
          "orderNumber": {"type": "hidden", "default": "42"},
          "productCode": {"type": "text"},
          "quantity": {"type": "number", "label": "quantity"}
        }
      },
      "uri": "http://api.x.io/orders/42/items",
      "label": "Add Item",
      "action": "append"
    }
  ]
}
    actual: |-
      {
  "h:link": [
    {
      "template": {
        "contentType": "application/x-www-form-urlencoded",
        "fields": {
          "orderNumber": {"type": "hidden", "default": "42"},
          "productCode": {"type": "text"},
          "quantity": {"type": "number", "label": "Quantity"}
        }
      },
      "uri": "http://api.x.io/orders/42/items",
      "label": "Add Item",
      "action": "append"
    }
  ]
}
```

**Comment by ljharb** [comment]
This is partially addressed. with #420.

**Comment by inadarei** [comment]
Correct. Now that #420 has been merged, I was able to implement the rest with https://github.com/inadarei/tap-better-diff ([tap-nirvana](https://www.npmjs.com/package/tap-nirvana) on NPM).

I believe implementing the formatting in a separate reporter is a better approach since everybody has their own preferences for how object should or should not be formatted and taking all of that into account in Tape itself will probably unreasonably increase the size of Tape.

I will submit a PR to add Tap Nirvana to  the list of the known reporters, on the README, if that is OK?
